src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/StorePal.cs

```diff
@@ -44,7 +44,47 @@ public static IExportPal LinkFromCertificateCollection(X509Certificate2Collectio
 
         public static IStorePal FromSystemStore(string storeName, StoreLocation storeLocation, OpenFlags openFlags)
         {
-            throw new NotImplementedException(nameof(FromSystemStore));
+            bool isReadWrite = (openFlags & OpenFlags.ReadWrite) == OpenFlags.ReadWrite;
+            if (isReadWrite && storeLocation == StoreLocation.LocalMachine)
+            {
+                // All LocalMachine stores are read-only
+                throw new CryptographicException(
+                    SR.Cryptography_Unix_X509_MachineStoresReadOnly,
+                    new PlatformNotSupportedException(SR.Cryptography_Unix_X509_MachineStoresReadOnly));
+            }
+
+            StringComparer ordinalIgnoreCase = StringComparer.OrdinalIgnoreCase;
+            switch (storeLocation)
+            {
+                case StoreLocation.CurrentUser:
+                {
+                    // Matches Unix behaviour of getting a disallowed store that is always empty.
+                    if (ordinalIgnoreCase.Equals(X509Store.DisallowedStoreName, storeName))
+                        return new UnsupportedDisallowedStore(openFlags);
+
+                    if (ordinalIgnoreCase.Equals(X509Store.RootStoreName, storeName))
+                    {
+                        if (isReadWrite)
+                            throw new CryptographicException(SR.Security_AccessDenied);
+
+                        return new TrustedStore(storeLocation);
+                    }
+                    break;
```

Comment:
Is `AndroidCAStore` just the roots, or does it also contain an intermediate CA cache?

Reply:
My understanding is it is just the roots. I didn't find anything exposing an intermediate CA cache (seems to be an implementation detail of the default trust manager).